### ISSUE TITLE
🍱 Improve clarity of QDMI flow diagram

### ISSUE THREAD

**Issue by echavarria-mqv** [Munich-Quantum-Software-Stack/QDMI#123]
<p align="center">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/Munich-Quantum-Software-Stack/QDMI/develop/docs/_static/qdmi_dark.svg" width="80%">
    <img src="https://raw.githubusercontent.com/Munich-Quantum-Software-Stack/QDMI/develop/docs/_static/qdmi.svg" width="80%">
  </picture>
</p>

I have always had some reservations about the diagram shown in the README file (and in flyers, posters, presentations, etc.). The loops going back to the beginning of both optimization stages make it seem like the design space exploration happens during the just-in-time compilation, but there are a few issues with that approach. I would suggest to remove the loops and present an acyclic diagram instead.

**Comment by burgholzer** [comment]
The original intention behind these back-loops is to indicate that compilation itself is _always_ an iterative procedure, where multiple passes are applied in sequence before moving to the next stage of compilation.
Technically, that picture is even more complicated. Copying a figure from our MQT Predictor paper here:
<img align="center" width="60%" alt="MQT Predictor Compilation Flow" src="https://github.com/user-attachments/assets/91365b99-f39e-4a55-8a64-9cf6680a6178">

That being said, at the abstraction-level we are trying to convey the execution flow in this overall picture, it is reasonable to remove these back-loops.
I briefly thought about simply moving the loops inside of the respective boxes, but this might not really provide too much value at this abstraction level.

TL;DR: we can remove the loops.
